File: AICodingWeb/Models/SupplierTransaction.cs
using System;

namespace AICodingWeb.Models
{
    public class SupplierTransaction
    {
        public int SupplierTransactionID { get; set; }
        public int SupplierID { get; set; }
        public int TransactionTypeID { get; set; }
        public int PurchaseOrderID { get; set; }
        public int PaymentMethodID { get; set; }
        public string SupplierInvoiceNumber { get; set; }
        public DateTime TransactionDate { get; set; }
        public decimal AmountExcludingTax { get; set; }
        public decimal TaxAmount { get; set; }
        public decimal TransactionAmount { get; set; }
        public decimal OutstandingBalance { get; set; }
        public DateTime? FinalizationDate { get; set; }
        public int LastEditedBy { get; set; }
        public DateTime LastEditedWhen { get; set; }

        public virtual Supplier Supplier { get; set; }
        public virtual TransactionType TransactionType { get; set; }
        public virtual PurchaseOrder PurchaseOrder { get; set; }
        public virtual PaymentMethod PaymentMethod { get; set; }
    }
} 

File: AICodingWeb/aicoding_web/app/models/supplier_transaction.py
from datetime import datetime
from app import db
from app.models.base import BaseModel

class SupplierTransaction(db.Model, BaseModel):
    """SupplierTransaction model"""
    
    __tablename__ = 'supplier_transactions'
    
    SupplierTransactionID = db.Column(db.Integer, primary_key=True)
    SupplierID = db.Column(db.Integer, db.ForeignKey('suppliers.SupplierID'), nullable=False)
    TransactionTypeID = db.Column(db.Integer, db.ForeignKey('transaction_types.TransactionTypeID'), nullable=False)
    PurchaseOrderID = db.Column(db.Integer, db.ForeignKey('purchase_orders.PurchaseOrderID'))
    PaymentMethodID = db.Column(db.Integer, db.ForeignKey('payment_methods.PaymentMethodID'))
    SupplierInvoiceNumber = db.Column(db.String(20))
    TransactionDate = db.Column(db.DateTime, nullable=False)
    AmountExcludingTax = db.Column(db.Numeric(18, 2), nullable=False)
    TaxAmount = db.Column(db.Numeric(18, 2), nullable=False)
    TransactionAmount = db.Column(db.Numeric(18, 2), nullable=False)
    OutstandingBalance = db.Column(db.Numeric(18, 2), nullable=False)
    FinalizationDate = db.Column(db.DateTime)
    LastEditedBy = db.Column(db.Integer, db.ForeignKey('people.PersonID'), nullable=False)
    LastEditedWhen = db.Column(db.DateTime, nullable=False)
    
    def __repr__(self):
        return f'<SupplierTransaction {self.SupplierTransactionID}>' 